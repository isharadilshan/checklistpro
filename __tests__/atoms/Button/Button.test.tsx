import React from 'react';
import {describe, expect} from '@jest/globals';
import {jest} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Button from '../../../src/components/atoms/Button';

// jest.useFakeTimers();

configure({adapter: new Adapter()});

const props = {
  label: 'TEST BUTTON',
  style: {width: 200},
  labelStyle: {backgroundColor: 'red'},
  contentStyle: {backgroundColor: 'red'},
  onPress: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<Button {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
