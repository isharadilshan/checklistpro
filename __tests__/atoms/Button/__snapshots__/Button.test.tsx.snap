// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<withTheme(Button)
  contentStyle={
    [
      {
        "height": 48,
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
  labelStyle={
    [
      {
        "fontSize": 20,
        "lineHeight": 22,
      },
      {
        "color": "white",
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
  mode="contained"
  onPress={[MockFunction]}
  style={
    [
      {
        "backgroundColor": "#007d7d",
        "borderColor": "#BBC8D6",
        "elevation": 0,
      },
      {
        "borderWidth": 0,
      },
      {
        "width": 200,
      },
    ]
  }
  uppercase={false}
>
  TEST BUTTON
</withTheme(Button)>
`;
